samples/java-customer-registry/src/main/java/customer/Main.java

```diff
@@ -50,12 +39,7 @@ public static void main(String[] args) throws Exception {
               "customerByName",
               CustomerDomain.getDescriptor())
           // end::register[]
-          .lowLevel()
-          .registerValueEntity(
-              context -> new CustomerValueEntityHandler(new CustomerValueEntity()),
-              CustomerValueEntityHandler.serviceDescriptor,
-              CustomerValueEntityHandler.entityType,
-              ValueEntityOptions.defaults())
+          .registerValueEntity(new CustomerValueEntityFactory())
```

Comment:
That's the main reason for this experiment. 

Because users should be able to inject whatever they want in the entity, they will end up having to touch the registration. With the previous API, they will have to be exposed to all the bits that are not so relevant and will have copy and paste them from the generate code to their main function. 

With the new factory, the registration only requires passing a factory and the user can inject whatever they want in the factory.